components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/internal/ServiceComponent.java

```diff
@@ -198,6 +214,124 @@ public Response saveExecutionPlan(String saveConfig) {
     }
 
     @GET
+    @Path("/workspace/root")
+    @Produces("application/json")
+    public Response root() {
+        try {
+            return Response.status(Response.Status.OK)
+                    .entity(workspace.listRoots())
+                    .header("Access-Control-Allow-Origin", '*')
```

Comment:
Can't we handle CORS without allowing any origin since this service is only supposed to access by our editor?